service-base/src/main/java/org/eclipse/hono/service/VertxBasedHealthCheckServer.java

```diff
@@ -128,17 +118,31 @@ public void setAdditionalResources(final List<Handler<Router>> resourceProviders
         registerAdditionalResources();
 
         final Future<Void> result = Future.future();
-        CompositeFuture.all(bindSecureHttpServer(), bindInsecureHttpServer())
-                .compose(r -> result.complete(), result);
+            bindSecureHttpServer().setHandler(res -> {
+                if (res.failed()){
+                    bindInsecureHttpServer().compose(r -> result.complete(), result);
+                } else if ( (config.isInsecurePortEnabled() && config.getInsecurePort(DEFAULT_PORT) != server.actualPort() )
+                        || (config.getInsecurePort() != Constants.PORT_UNCONFIGURED && config.getInsecurePort() != server.actualPort())) {
+                    bindInsecureHttpServer().compose(r -> result.complete(), result);
+                } else {
+                    result.complete();
+                }
+            });
         return result;
     }
 
     private Future<Void> bindInsecureHttpServer() {
-        if (config.isInsecurePortEnabled()) {
-            final Future<Void> result = Future.future();
+        final Future<Void> result = Future.future();
 
+        if (config.getInsecurePortBindAddress() == Constants.LOOPBACK_DEVICE_ADDRESS) {
+            LOG.warn("Won't start insecure health checks HTTP server: no bind address configured.");
+            return Future.failedFuture("NOT_ENABLED");
+        } else if (config.isInsecurePortEnabled() || config.getInsecurePort() == Constants.PORT_UNCONFIGURED ) {
```

Comment:
it would be easier to read if we would check for the condition that actually allows to start the server:
```
else if (config.isInsecurePortEnabled() || config.getInsecurePort() != Constants.PORT_UNCONFIGURED) {
  // start server
} else {
  // log error
}
```
In any case, the current check is wrong as it doesn't start the server if the *insecurePortEnabled* flag is set to `true` ...

Reply:
thanks for catching that. It's probably a bad copy and paste..